rundetection/rules/loq_rules.py

```diff
@@ -0,0 +1,174 @@
+    reversed_files = reversed(sans_files)
+    for sans_file in reversed_files:
+        if _is_sample_direct_file(sans_file=sans_file):
+            return sans_file
+    return None
+
+
+def _find_can_scatter_file(sans_files: list[SansFileData], can_title: str) -> SansFileData | None:
+    for sans_file in sans_files:
+        if _is_can_scatter_file(sans_file=sans_file, can_title=can_title):
+            return sans_file
+    return None
+
+
+def _find_can_trans_file(sans_files: list[SansFileData], can_title: str) -> SansFileData | None:
+    for sans_file in sans_files:
+        if _is_can_transmission_file(sans_file=sans_file, can_title=can_title):
+            return sans_file
+    return None
+
+
+def find_path_for_run_number(cycle_path: str, run_number: int) -> Path | None:
+    # 10 is just a magic number, but we needed an unrealistic value for the maximum
+    for padding in range(1, 11):
+        potential_path = Path(f"{cycle_path}/LOQ{str(run_number).zfill(padding)}.nxs")
+        if potential_path.exists():
+            return potential_path
+    return None
+
+
+def grab_cycle_instrument_index(cycle: str) -> str:
+    _, cycle_year, cycle_num = cycle.split("_")
+    url = f"http://data.isis.rl.ac.uk/journals/ndxloq/journal_{cycle_year}_{cycle_num}.xml"
+    return requests.get(url, timeout=5).text
+
+
+def create_list_of_files(job_request: JobRequest) -> list[SansFileData]:
+    cycle = _extract_cycle_from_file_path(file_path=job_request.filepath)
+    xml = grab_cycle_instrument_index(cycle=cycle)
+    cycle_run_info = xmltodict.parse(xml)
+    list_of_files = []
+    for run_info in cycle_run_info["NXroot"]["NXentry"]:
+        title_contents = run_info["title"]["#text"].split("_")
+        run_number = run_info["run_number"]["#text"]
+        if len(title_contents) == 2:  # noqa: PLR2004
+            _, file_type = title_contents
+        elif len(title_contents) == 3:  # noqa: PLR2004
+            _, __, file_type = title_contents
+        else:
+            job_request.will_reduce = False
+            logger.error(f"Run {run_info} either doesn't contain a _ or is not an expected experiment title format.")
+            return []
+        list_of_files.append(SansFileData(title=run_info["title"]["#text"], type=file_type, run_number=run_number))
+    return list_of_files
+
+
+def strip_excess_files(sans_files: list[SansFileData], scatter_run_number: int) -> list[SansFileData]:
+    new_list_of_files: list[SansFileData] = []
+    for sans_file in sans_files:
+        if int(sans_file.run_number) >= scatter_run_number:
+            return new_list_of_files
+        new_list_of_files.append(sans_file)
+    return new_list_of_files
+
+
+class LoqFindFiles(Rule[bool]):
+    def __init__(self, value: bool):
+        super().__init__(value)
+
+    def verify(self, job_request: JobRequest) -> None:
+        # Expecting 3 values
+        if len(job_request.experiment_title.split("_")) != 3:  # noqa: PLR2004
```

Comment:
`job_request.experiment_title.split("_")` gets split more than once in this func so maybe make a new variable for it